### PR TITLE
Read original issuer from Scoping

### PR DESCRIPTION
**Pull request by MKodde**
Gateway will set scoping on the authnrequest, it contains the SP where the original authnrequest originated from.

This value is used to judge if this is an RA authentication (vetting procedure). If this is the case, the ForceAuthn is added to the outgoing request.

This is deemed POC code and should be reviewed carefully, any additional defence in-depth changes are welcome but might be postponed until the team decides how to move forward with the POC.

See:
https://www.pivotaltracker.com/story/show/171101458
https://github.com/OpenConext/Stepup-gssp-bundle/pull/33
https://github.com/OpenConext/Stepup-saml-bundle/pull/97

### CODE REVIEW

**Comment by MKodde** [comment]
The build breaks on a JS security warning. I'm addressing this issue in another PR.

File: config/services.yaml [hunk 1]
```diff
@@ -30,5 +30,4 @@ services:
 
     Surfnet\AzureMfa\Application\Service\AuthenticationHelper:
         arguments:
-            $regex: '%ra_issuer_entity_id_regex%'
-            $authenticationService: '@Surfnet\GsspBundle\Service\AuthenticationService'
\ No newline at end of file
+            $regex: '%ra_issuer_entity_id_regex%'
```
Comment (pablothedude) [inline config/services.yaml]: Consider not using a regex for KISS and DID

File: src/Surfnet/AzureMfa/Infrastructure/Controller/DefaultController.php [hunk 2]
```diff
@@ -148,7 +151,7 @@ public function authenticationAction(Request $request)
         $user = $this->azureMfaService->startAuthentication(new UserId($nameId));
 
         return new RedirectResponse(
-            $this->azureMfaService->createAuthnRequest($user, $this->authenticationHelper->useForceAuthn())
+            $this->azureMfaService->createAuthnRequest($user, $this->authenticationHelper->useForceAuthn($request))
```
Comment (pablothedude) [inline src/Surfnet/AzureMfa/Infrastructure/Controller/DefaultController.php]: It seems that the AuthenticationHelper::useForceAuthn method doesn't want the parameter?

File: src/Surfnet/AzureMfa/Infrastructure/Controller/DefaultController.php [hunk 3]
```diff
@@ -28,6 +28,9 @@
 use Surfnet\GsspBundle\Service\AuthenticationService;
 use Surfnet\GsspBundle\Service\RegistrationService;
 use Surfnet\SamlBundle\Http\Exception\AuthnFailedSamlResponseException;
+use Surfnet\SamlBundle\Http\RedirectBinding;
```
Comment (pablothedude) [inline src/Surfnet/AzureMfa/Infrastructure/Controller/DefaultController.php]: Are all added imports really needed?

File: config/services.yaml (hunk 1 above)
Comment (MKodde) [reply config/services.yaml]: Not something to fix here in my opinion. And for now this is what PM and I came up with.

File: src/Surfnet/AzureMfa/Infrastructure/Controller/DefaultController.php (hunk 2 above)
Comment (MKodde) [reply src/Surfnet/AzureMfa/Infrastructure/Controller/DefaultController.php]: Good call, removed the unwarranted paramter from controller & tests

File: src/Surfnet/AzureMfa/Infrastructure/Controller/DefaultController.php (hunk 3 above)
Comment (MKodde) [reply src/Surfnet/AzureMfa/Infrastructure/Controller/DefaultController.php]: 15 imports checken, none where removed

**Pull request merged**